docs/models/using.md

```diff
@@ -0,0 +1,133 @@
+
+##### `ignore`
+
+This is also fairly self-explanatory. The `IStateMapper` will simply ignore the given properties when mapping a blockstate. Note that calling `ignore` twice will simply combine the two lists, and that a blockstate without one of the ignored properties will not cause an error in the mapper.
+
+Using Item Models
+-----------------
+
+Unlike blocks, which automatically have a default `IStateMapper` that works without any extra registration, items must be registered to their models manually. Thankfully, this is very easy. Simply call `ModelLoader.setCustomModelResourceLocation`. This method takes the item and metadata we're registering a model for, and an MRL that points to a model. The way the game searches for the corresponding file is as follows:
+
+1. For an MRL `<domain>:<path>#<varstr>`
+2. Check blockstate JSON at `assets/<domain>/blockstates/<path>.json` for the variant `<varstr>`, and use that if available.
+3. If that fails, check that `varstr.equals("inventory")`. (Vanilla specialcasing)
+4. If that succeeds, find the item model at `assets/<domain>/models/item/<path>`.
+
+When using a JSON item model from `models/item`, you can also leverage [overrides][].
+
+!!! note
+    When you call `ModelLoader.setCustomModelResourceLocation`, it also calls `ModelBakery.registerItemVariants` with the item and MRL given. This sets up the model for baking later.
+
+### `ItemMeshDefinition`
+
+An `ItemMeshDefinition` is a function that takes `ItemStack`s and maps them to `ModelResourceLocation`s. They are registered per item, and you can do so with `ModelLoader.setCustomMeshDefinition`, which takes an item and the `ItemMeshDefinition` to use for its `ItemStack`s.
+
+!!! important
+    When you call `ModelLoader.setCustomMeshDefinition`, it **does not** call `ModelBakery.registerItemVariants`. For each and every MRL your `ItemMeshDefinition` can return, you must call this method in order for it to work.
+
+### Blockstate JSONs for Items
+
+Note that *items* can use *block*state JSONs. This is possible by simply passing an MRL pointing to a blockstate JSON into `ModelLoader.setCustomeModelResourceLocation` or returning it from an `ItemMeshDefinition`. Doing so allows you to take advantage of things like submodels (and combining variants if you use the Forge format) for item models. In the Forge format, this is especially useful for the default item layer model as you can use the `textures` block inside combining variant definitions to build up the layers of the model. Note that there is a major caveat. First, blockstate JSONs can only refer to models under `models/block`; they cannot see models under `models/item`. This means that you can't use `minecraft:item/generated` to get the default item transforms (proper rotation and scale). In a Forge blockstate JSON, you can use the model `minecraft:builtin/generated` and set the `transform` property to `forge:default-item` to work around this. Here's an example:
```

Comment:
this is a mouthful esp if you're a beginner. try breaking it up a bit or removing/reducing some of the fluff about builtin/generated